projecthub/src/main/java/com/project/hub/auth/oauth/handler/OAuth2SuccessHandler.java

```diff
@@ -0,0 +1,69 @@
+package com.project.hub.auth.oauth.handler;
+
+import com.project.hub.auth.service.TokenComponent;
+import com.project.hub.repository.UserRepository;
+import com.project.hub.model.type.UserRole;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.web.authentication.SimpleUrlAuthenticationSuccessHandler;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+import com.project.hub.entity.User;
+
+@Slf4j
+@Component
+public class OAuth2SuccessHandler extends SimpleUrlAuthenticationSuccessHandler {
+
+  private final TokenComponent tokenComponent;
+  private final UserRepository userRepository;
+
+  public OAuth2SuccessHandler(
+      TokenComponent tokenComponent,
+      UserRepository userRepository) {
+    this.tokenComponent = tokenComponent;
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  public void onAuthenticationSuccess(HttpServletRequest request, HttpServletResponse response
+      , Authentication authentication) throws IOException {
+
+    OAuth2User oAuth2User = (OAuth2User) authentication.getPrincipal();
+    String email = oAuth2User.getAttribute("email");
+
+    Optional<User> user = userRepository.findByEmail(email);
+    Long userId;
+    String targetUrl;
+    log.info("OAuth2User: {}", oAuth2User);
+    if (user.isPresent()) { // 기존 회원인 경우 액세스, 리프레시 토큰 생성 후 전달
+      userId = user.get().getId();
+
+    } else { // 신규 회원인 경우 회원가입 페이지로 이동
+      User newOAuth2User = User.builder()
+          .email(email)
+          .nickname(oAuth2User.getAttribute("name"))
+          .role(UserRole.USER)
+          .provider("google")
+          .providerId(oAuth2User.getAttribute("sub"))
+          .build();
+      User savedOAuthUser = userRepository.save(newOAuth2User);
+      userId = savedOAuthUser.getId();
+    }
```

Comment:
개발하실때 Optional type의 variable에 대해서 .get()을 부르는 것을 가능하면 지양해주시면 좋을 것 같습니다. 안티패턴이라고 생각하시면 좋을 것 같아요. 
```suggestion
    log.info("OAuth2User: {}", oAuth2User);
    Long userId = userRepository.findByEmail(email).map(user -> user.getId()).orElseGet(() -> {
        User newOAuth2User = User.builder()
          .email(email)
          .nickname(oAuth2User.getAttribute("name"))
          .role(UserRole.USER)
          .provider("google")
          .providerId(oAuth2User.getAttribute("sub"))
          .build();
      User savedOAuthUser = userRepository.save(newOAuth2User);
      return  savedOAuthUser.getId();
      });
```

Reply:
repository에서 찾을 때 User로 찾고 orElseThrow로 예외 처리하겠습니다!